WordPress/src/main/java/org/wordpress/android/ui/posts/PostResolutionOverlayFragment.kt

```diff
@@ -0,0 +1,402 @@
+        1f
+    } else {
+        ContentAlpha.medium
+    }
+
+@Suppress("DEPRECATION")
+class PostResolutionOverlayFragment : BottomSheetDialogFragment() {
+    @Inject
+    internal lateinit var viewModelFactory: ViewModelProvider.Factory
+
+    private lateinit var viewModel: PostResolutionOverlayViewModel
+
+    internal lateinit var listener: PostResolutionOverlayListener
+
+    private val postModel: PostModel? by lazy {
+        arguments?.getSerializable(ARG_POST_MODEL) as? PostModel
+    }
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        if (context is PostResolutionOverlayListener) {
+            listener = context
+        } else {
+            throw RuntimeException("$context must implement PostResolutionOverlayListener")
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (requireNotNull(activity).application as WordPress).component().inject(this)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        return super.onCreateDialog(savedInstanceState).apply {
+            (this as? BottomSheetDialog)?.fillScreen(isDraggable = true)
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        initializeViewModelAndStart()
+        return ComposeView(requireContext()).apply {
+            setContent {
+                AppTheme {
+                    val uiState by viewModel.uiState.observeAsState()
+                    PostResolutionOverlay(uiState)
+                }
+            }
+        }
+    }
+    private fun initializeViewModelAndStart() {
+        viewModel = ViewModelProvider(this, viewModelFactory)[PostResolutionOverlayViewModel::class.java]
+
+        viewModel.triggerListeners.observe(viewLifecycleOwner) {
+            // todo: annmarie - this need to be changed once the data class is created
+            listener.onOverlayAction(PostResolutionOverlayAction.TO_BE_DETERMINED)
+        }
+
+        viewModel.dismissDialog.observe(viewLifecycleOwner) {
+            dismiss()
+        }
+
+        viewModel.start(postModel)
+    }
+
+    override fun onDismiss(dialog: DialogInterface) {
+        super.onDismiss(dialog)
+        Log.i(javaClass.simpleName, "***=> onDismiss captured")
+        viewModel.onDialogDismissed()
+    }
+
+    companion object {
+        const val TAG = "PostResolutionOverlayFragment"
+
+        // todo: annmarie maybe not send the entire post model
+        private const val ARG_POST_MODEL = "arg_post_model"
+
+        // todo: annmarie will the parameters require not null - should we not worry about that????
+        fun newInstance(postModel: PostModel) = PostResolutionOverlayFragment().apply {
+            arguments = Bundle().apply {
+                putSerializable(ARG_POST_MODEL, postModel)
+            }
+        }
+    }
+
+    @Composable
```

Comment:
@zwarm a really minor suggestion here would be to move the composables to a different file with a name like PostResolutionComposables or PostResolutionScreen. This might improve the readability. This is just a matter of personal preference so please feel free to ignore it. :-)

Reply:
Yes, this is already on my list of things to do. The final cleanup will take care of it.